api/base/serializers.py

```diff
@@ -407,10 +414,32 @@ def to_internal_value(self, data):
             if request.method in utils.UPDATE_METHODS and not utils.is_bulk_request(request):
                 id_field = self.get_id(self.root.instance)
                 if id_field != data:
-                    raise api_exceptions.Conflict(detail=('The id you used in the URL, "{}", does not match the id you used in the json body\'s id field, "{}". The object "{}" exists, otherwise you\'d get a 404, so most likely you need to change the id field to match.'.format(id_field, data, id_field)))
+                    raise api_exceptions.Conflict(
+                        detail=(
+                            f'The id you used in the URL, "{id_field}", does not match the id you used in the json'
+                            f' body\'s id field, "{data}". The object "{id_field}" exists, otherwise you\'d get a'
+                            f' 404, so most likely you need to change the id field to match.'
+                        ),
+                    )
         return super(IDField, self).to_internal_value(data)
 
     def get_id(self, obj):
+        if isinstance(obj, BaseFileNode) and self.source == '_id':
+            guid = Guid.load(self.context['view'].kwargs['file_id'])
+            if guid:
+                return guid._id
+
+        return getattr(obj, self.source, '_id')
+
+
+class GuidOrIDField(IDField):
+
+    def get_id(self, obj):
```

Comment:
am I missing something or is this identical to `get_id` in the superclass? If so, either revert that or delete this.

Reply:
Yes, I'm sorry that's just an mistake.